include/Megamix/Types.hpp

```diff
@@ -6,278 +6,479 @@
+    this->a = 0;
+  }
+
+  Color8(u8 r, u8 g, u8 b, u8 a) {
+    this->r = r;
+    this->g = g;
+    this->b = b;
+    this->a = a;
+  }
+
+  Color8(u32 color) {
+    this->r = color >> 24;
+    this->g = color >> 16;
+    this->b = color >> 8;
+    this->a = color;
+  }
+};
+
+struct MuseumRowColor {
+  Color8 unk1;
+  Color8 background;
+  Color8 edgeFade;
+  Color8 unk3;
+
+  MuseumRowColor(Color8 background, Color8 edgeFade) {
+    this->unk1 = {0x00, 0x00, 0x00, 0x00};
+    this->unk3 = {0xFF, 0xFF, 0xFF, 0x6E};
+
+    this->background = background;
+    this->edgeFade = edgeFade;
+  }
+};
+
+struct CList {
+  int *_vtable;
+  struct CList *mNext;
+  struct CList *mPrev;
+};
+
+struct TickFlowStack {
+  u32 *tick;
+  u32 tickSize;
+};
+
+struct CTickflow {
+  CList at0;
+  u32 *mTickflow;
+  u32 mCategory;
+  int mTickFlowSize;
+  float mRest;
+  u8 unk1;
+  u32 mCondvar;
+  u32 mCondvarStack[10];
+  u8 pad1[0x17];
+  u8 mCondvarStackCounter;
+  struct TickFlowStack mTickFlowStack[8];
+  u8 pad2[0x3F];
+  u8 mTickFlowStackCounter;
+  u8 mButtonPromptIsReleased;
+  u8 mButtonPromptIsPressed;
+  u32 mButtonPromptButton;
+  int mButtonPromptPressSfx;
+  int mButtonPromptReleaseSfx;
+  u8 pad3[7];
+  u32 mCountdownValue;
+  u32 mCountdownCounter;
+};
+
+//-----------------
+//Save File
+//-----------------
+
+typedef enum EGameRank {
+  eLocked = 0,
+  eUnplayed = 1,
+  eUnfinished = 2,
+  eNot_Good = 3,
+  eOk = 4,
+  eHigh = 5,
+  ePerfect = 6
+} EGameRank;
+
+typedef enum EPlayStyle { /* simple tap / button */
+  ePlayStyle_Button = 0,
+  ePlayStyle_Tap = 1
+} EPlayStyle;
+
+struct CSaveData_Sub2 {
+  u8 pad1[0xF];
+  u32 mPlayTime;
+  u8 mPlayerData; /* Created by retype action */
+  u8 pad2[0x5F];
+  enum EGameRank mGameRanks[104];
+  u8 pad3[0xF18];
+  u8 field3957_0xff4[1][16];
+  u8 pad4[0xBF];
+  u8 mLocale; /* Created by retype action */
+  u8 pad5[0x91];
+  bool mUnlockedGoat; /* Created by retype action */
+  u8 pad6[0x5];
+  u8 mIsFirstMuseumTalk; /* Created by retype action */
+  u8 pad7[0x2];
+  bool hasBeenInCafe;
+  u8 pad8[0x1D];
+  u16 mClPlay2;
+  u16 mCLPlay3;
+  u8 pad9[0x2];
+  u16 mMoney;
+  u16 mCoin;
+  enum EPlayStyle mPlayStyle;
+  u8 mTimingEft;
+  u8 isCec; /* Created by retype action */
+  u8 pad10[0x131];
+  u8 field4646_0x12bc[104];
+  u8 pad11[0xAA];
+  u8 mOnion[3];
+  u8 pad12[0x7];
+  u32 mGoatLv;
+  u8 pad13[0x26B];
+};
+
+struct CSaveData {
+  int *_vtable;
+  u32 unk1;
+  u8 unk2;
+  u8 unk3;
+  u32 unk4;
+  u32 unk5[4];
+  u8 cSaveData_sub1[0x1c20];
+  struct CSaveData_Sub2 mFileData[4];
+  u32 mCurrentFile;
+};
+
+//-----------------
+//Input Manager
+//-----------------
+
+typedef enum EButtonFlag {
+    eButton_None=0,
+    eButton_A=1,
+    eButton_B=2,
+    eButton_Select=4,
+    eButton_Start=8,
+    eButton_DpadRight=16,
+    eButton_DpadLeft=32,
+    eButton_DpadUp=64,
+    eButton_DpadDown=128,
+    eButton_R=256,
+    eButton_L=512,
+    eButton_X=1024,
+    eButton_Y=2048
+} EButtonFlag;
```

Comment:
`typedef enum` -> `enum class` (for scope reasons)
`eButton_X` -> `X`
`4` -> `1 << 2`